File: matiasmasso/Mat.NET/DT2/DTOBrowser.vb
Imports System.Web
Imports Common.Logging.Configuration

Public Class DTOBrowser
    Property IP As String
    Property OS As String
    Property Browser As String
    Property UserName As String
    Property SessionId As String
    Property Rol As String

    Property Cookies As Dictionary(Of String, String)
    Property Headers As Dictionary(Of String, String)

    Shared Function Factory(IP As String, BrowserName As String, BrowserVersion As String, oUser As DTOUser) As DTOBrowser
        Dim retval As New DTOBrowser
        With retval
            .IP = IP
            .Browser = String.Format("{0} {1}", BrowserName, BrowserVersion)
            If oUser IsNot Nothing Then
                .UserName = DTOUser.NicknameOrElse(oUser)
                .Rol = oUser.Rol.id.ToString
            End If
        End With
        Return retval
    End Function

    Shared Function EmailLink(oBrowser As DTOBrowser, BlScriptsEnabled As Boolean) As String
        Dim sb As New System.Text.StringBuilder
        For Each s As String In List(oBrowser, BlScriptsEnabled)
            sb.AppendLine(s)
        Next

        Dim src As String = sb.ToString
        Dim sBody As String = HttpUtility.UrlEncode(src.ToString())
        Dim sSubject As String = HttpUtility.UrlEncode("prueba de navegador de usuario")
        Dim sTo As String = "matias@matiasmasso.es"
        Dim retval As String = "mailto:" & sTo & "?subject=" & sSubject & "&body=" & sBody
        Return retval
    End Function

    Shared Function List(oBrowser As DTOBrowser, BlScriptsEnabled As Boolean) As List(Of String)
        Dim retval As New List(Of String)
        With oBrowser
            retval.Add("IP: " & .IP)
            'retval.Add("OS: " & .OS)
            retval.Add("Browser: " & .Browser)
            retval.Add("Scripts: " & If(BlScriptsEnabled, "SI", "NO"))
            retval.Add("User name: " & .UserName)
            retval.Add("Session: " & .SessionId)
            retval.Add("Rol: " & .Rol)
        End With
        Return retval
    End Function


End Class

File: matiasmasso/Mat.NET/DTO/DTOBrowser.cs
using System.Collections.Generic;
using System.Web;

namespace DTO
{
    public class DTOBrowser
    {
        public string IP { get; set; }
        public string OS { get; set; }
        public string Browser { get; set; }
        public string UserName { get; set; }
        public string SessionId { get; set; }
        public string Rol { get; set; }

        public Dictionary<string, string> Cookies { get; set; }
        public Dictionary<string, string> Headers { get; set; }

        public static DTOBrowser Factory(string IP, string BrowserName, string BrowserVersion, DTOUser oUser)
        {
            DTOBrowser retval = new DTOBrowser();
            {
                var withBlock = retval;
                withBlock.IP = IP;
                withBlock.Browser = string.Format("{0} {1}", BrowserName, BrowserVersion);
                if (oUser != null)
                {
                    withBlock.UserName = DTOUser.NicknameOrElse(oUser);
                    withBlock.Rol = oUser.Rol.id.ToString();
                }
            }
            return retval;
        }

        public static string EmailLink(DTOBrowser oBrowser, bool BlScriptsEnabled)
        {
            System.Text.StringBuilder sb = new System.Text.StringBuilder();
            foreach (string s in List(oBrowser, BlScriptsEnabled))
                sb.AppendLine(s);

            string src = sb.ToString();
            string sBody = HttpUtility.UrlEncode(src.ToString());
            string sSubject = HttpUtility.UrlEncode("prueba de navegador de usuario");
            string sTo = "matias@matiasmasso.es";
            string retval = "mailto:" + sTo + "?subject=" + sSubject + "&body=" + sBody;
            return retval;
        }

        public static List<string> List(DTOBrowser oBrowser, bool BlScriptsEnabled)
        {
            List<string> retval = new List<string>();
            {
                var withBlock = oBrowser;
                retval.Add("IP: " + withBlock.IP);
                // retval.Add("OS: " & .OS)
                retval.Add("Browser: " + withBlock.Browser);
                retval.Add("Scripts: " + (BlScriptsEnabled ? "SI" : "NO"));
                retval.Add("User name: " + withBlock.UserName);
                retval.Add("Session: " + withBlock.SessionId);
                retval.Add("Rol: " + withBlock.Rol);
            }
            return retval;
        }
    }
}
